insights/src/main/java/org/jboss/as/telemetry/extension/InsightsService.java

```diff
@@ -0,0 +1,490 @@
+        } catch (IOException e) {
+            ROOT_LOGGER.couldNotCreateEditPropertiesFile(e);
+        } finally {
+            if (fileOut != null) {
+                try {
+                    fileOut.close();
+                } catch (IOException e) {
+                    ROOT_LOGGER.couldNotClosePropertiesFile(e);
+                }
+            }
+            if (fis != null) {
+                try {
+                    fis.close();
+                } catch (IOException e) {
+                    ROOT_LOGGER.couldNotClosePropertiesFile(e);
+                }
+            }
+        }
+        setRhnLoginCredentials(rhnUid, rhnPw);
+    }
+
+    public void setRhnLoginCredentials(String rhnUid, String rhnPw,
+            String proxyUrl, String proxyPort, String proxyUser, String proxyPwd) {
+        Properties properties = new Properties();
+        FileOutputStream fileOut = null;
+        FileInputStream fis = null;
+        File file = getPropertiesFile();
+        try {
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+            } else {
+                fis = new FileInputStream(file.getPath());
+                properties.load(fis);
+            }
+            properties.setProperty(InsightsExtension.PROXY_PASSWORD, proxyPwd);
+            properties.setProperty(InsightsExtension.PROXY_USER, proxyUser);
+            fileOut = new FileOutputStream(file);
+            properties.store(fileOut, TELEMETRY_DESCRIPTION);
+        } catch (IOException e) {
+            ROOT_LOGGER.couldNotCreateEditPropertiesFile(e);
+        } finally {
+            if (fileOut != null) {
+                try {
+                    fileOut.close();
+                } catch (IOException e) {
+                    ROOT_LOGGER.couldNotClosePropertiesFile(e);
+                }
+            }
+            if (fis != null) {
+                try {
+                    fis.close();
+                } catch (IOException e) {
+                    ROOT_LOGGER.couldNotClosePropertiesFile(e);
+                }
+            }
+        }
+        setRhnLoginCredentials(rhnUid, rhnPw, proxyUrl, proxyPort);
+    }
+
+    public void setConnectionManager() {
+        String username = "";
+        String password = "";
+        String url = "";
+        String proxyUser = null;
+        String proxyPassword = null;
+        URL proxyUrl = null;
+        int proxyPort = 0;
+        String userAgent = null;
+        String jbossConfig = System.getProperty(JBOSS_PROPERTY_DIR);
+        String propertiesFilePath = jbossConfig + File.separator
+                + InsightsExtension.SUBSYSTEM_NAME + File.separator
+                + TELEMETRY_PROPERTY_FILE_NAME;
+        Properties properties = new Properties();
+        FileInputStream fis = null;
+        File file = new File(propertiesFilePath);
+        if (file.exists()) {
+            try {
+                fis = new FileInputStream(propertiesFilePath);
```

Comment:
it would be easier if you would use try with resources here.
try (FileInputStream fis  = new FileInputStream(propertiesFilePath){
this way you don't need to manually close the stream